Source/WebCore/css/CSSRule.cpp

```diff
@@ -70,6 +72,18 @@ bool CSSRule::hasStyleRuleAncestor() const
     return false;
 }
 
+CSSParserEnum::IsNestedContext CSSRule::nestedContext() const
+{
+    for (RefPtr parentRule = this->parentRule(); parentRule; parentRule = parentRule->parentRule()) {
+        if (is<CSSStyleRule>(parentRule))
+            return CSSParserEnum::AncestorRuleType::Style;
+        if (is<CSSScopeRule>(parentRule))
+            return CSSParserEnum::AncestorRuleType::Scope;
```

Comment:
```suggestion
        if (is<CSSStyleRule>(*parentRule))
            return CSSParserEnum::AncestorRuleType::Style;
        if (is<CSSScopeRule>(*parentRule))
            return CSSParserEnum::AncestorRuleType::Scope;
```

This reduces a null check I believe.